keras/metrics.py

```diff
@@ -71,6 +71,25 @@ def cosine_proximity(y_true, y_pred):
     return -K.mean(y_true * y_pred)
 
 
+def matthews(y_true, y_pred):
+    ''' Matthews correlation coefficient
+    '''
+    y_pred_pos = K.round(y_pred)
+    y_pred_neg = 1 - y_pred_pos
```

Comment:
The targets / predictions are assumed to be in the [0, 1] interval but that is not enforced. It should be.